cmds/minigames/connect4.js

```diff
@@ -175,288 +289,320 @@ class connect4 extends two_p_Game{
-                this.map[y][x] = "🟡"
-            }else{
-                this.map[y][x] = "🔴"
+        if (this.stack[col_i] !== -1) {
+            x = col_i;
+            y = this.stack[col_i];
+            if (color === "yellow") {
+                this.map[y][x] = "🟡";
+            } else {
+                this.map[y][x] = "🔴";
             }
-            this.stack[col_i] -= 1
-
-        }else{
-
+            this.stack[col_i] -= 1;
+        } else {
+            // invalid move - column is full
             return 0;
-
         }
 
-        return this.check(x, y)
-
+        return this.check(x, y);
     }
 
-    check(x, y, board = this.map){
-
-        let color = board[y][x]
+    // check if anyone wins
+    check(x, y, board = this.map) {
+        let color = board[y][x];
         if (color === "⚪") return 1;
         //horizontal
-        if(this.search(x, y , color, 1, 0, 0) + this.search(x, y , color, -1, 0, 0) - 1 >= 4){
-            return 2
+        if (
+            this.search(x, y, color, 1, 0, 0) +
+            this.search(x, y, color, -1, 0, 0) - 1 >= 4
+        ) {
+            return 2;
         }
         //lt-rb diagonal
-        if(this.search(x, y , color, 1, 1, 0) + this.search(x, y , color, -1, -1, 0) - 1 >= 4){
-            return 2
+        if (
+            this.search(x, y, color, 1, 1, 0) +
+            this.search(x, y, color, -1, -1, 0) - 1 >= 4
+        ) {
+            return 2;
         }
         //lt-rb diagonal
-        if(this.search(x, y , color, -1, 1, 0) + this.search(x, y , color, 1, -1, 0) - 1 >= 4){
-            return 2
+        if (
+            this.search(x, y, color, -1, 1, 0) +
+            this.search(x, y, color, 1, -1, 0) - 1 >= 4
+        ) {
+            return 2;
         }
         //vertical
-        if(this.search(x, y , color, 0, 1, 0) + this.search(x, y , color, 0, -1, 0) - 1 >= 4){
-            return 2
+        if (
+            this.search(x, y, color, 0, 1, 0) +
+            this.search(x, y, color, 0, -1, 0) - 1 >= 4
+        ) {
+            return 2;
         }
 
-        let countstack = 0
-        for(let i = 0; i < 7; i++){
-            if(this.stack[i] === -1){
-                countstack++
+        // check how many pieces in a column
+        let countstack = 0;
+        for (let i = 0; i < 7; i++) {
+            if (this.stack[i] === -1) {
+                countstack++;
             }
         }
 
-        if(countstack === 7) return 3;
-
-        return 1
+        // a column is full
+        if (countstack === 7) return 3;
 
+        return 1;
     }
 
-    
-
-    search(x, y, color, dx, dy, count, board = this.map){
-
-        if(x<0 || x>6 || y<0 || y>5) return count
-        if(color !== board[y][x]) return count
-
-        return this.search(x+dx, y+dy, color, dx, dy, count+1, board)
+    // (not sure what this whole code does)
+    search(x, y, color, dx, dy, count, board = this.map) {
+        if (x < 0 || x > 6 || y < 0 || y > 5) return count;
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/xiaomiao-fong/Chen_bot/pullRequest?prid=7807535)
```suggestion
        if (x < 0 || x > 6 || y < 0 || y > 5) {return count;}
```